File: server/src/main/java/gg/mineral/server/network/packet/play/clientbound/EntityMetadataPacket.java
package gg.mineral.server.network.packet.play.clientbound;

import java.util.List;

import gg.mineral.api.network.packet.Packet;
import gg.mineral.server.entity.meta.EntityMetadataImpl;
import io.netty.buffer.ByteBuf;

public final record EntityMetadataPacket(int entityId, List<EntityMetadataImpl.Entry> entries)
        implements Packet.OUTGOING {
    @Override
    public void serialize(ByteBuf os) {
        os.writeInt(entityId);
        writeMetadata(os, entries);
    }

    @Override
    public byte getId() {
        return 0x1C;
    }
}

File: server/src/main/kotlin/gg/mineral/server/network/packet/play/clientbound/EntityMetadataPacket.kt
package gg.mineral.server.network.packet.play.clientbound

import gg.mineral.api.entity.meta.EntityMetadata
import gg.mineral.api.network.packet.Packet
import io.netty.buffer.ByteBuf

@JvmRecord
data class EntityMetadataPacket(val entityId: Int, val entries: List<EntityMetadata.Entry>) :
    Packet.OUTGOING {
    override fun serialize(os: ByteBuf) {
        os.writeInt(entityId)
        os.writeMetadata(entries)
    }

    override val id: Byte
        get() = 0x1C
}
